src/IssueViz.Security.UnitTests/OpenInIDE/Api/OpenInIDERequestHandlerTests.cs

```diff
@@ -0,0 +1,98 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Api;
+using SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Contract;
+using SonarQube.Client;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.UnitTests.OpenInIDE.Api
+{
+    [TestClass]
+    public class OpenInIDERequestHandlerTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            // Arrange
+            var validatorExport = MefTestHelpers.CreateExport<IOpenInIDEStateValidator>(Mock.Of<IOpenInIDEStateValidator>());
+            var sonarQubeServiceExport = MefTestHelpers.CreateExport<ISonarQubeService>(Mock.Of<ISonarQubeService>());
+            var loggerExport = MefTestHelpers.CreateExport<ILogger>(Mock.Of<ILogger>());
+
+            // Act & Assert
+            MefTestHelpers.CheckTypeCanBeImported<OpenInIDERequestHandler, IOpenInIDERequestHandler>(null, new[] { validatorExport, sonarQubeServiceExport, loggerExport });
+        }
+
+        [TestMethod]
+        public void ShowHotspot_InvalidArg_Throws()
+        {
+            IOpenInIDERequestHandler testSubject = new OpenInIDERequestHandler(Mock.Of<IOpenInIDEStateValidator>(), Mock.Of<ISonarQubeService>(), Mock.Of<ILogger>());
+
+            Action act = () => testSubject.ShowHotspot(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("request");
+        }
+
+        [TestMethod]
+        public void ShowHotpot_IdeNotInCorrectState_NoActionTaken()
+        {
+            var validatorMock = CreateValidator("http://localhost/", "project", "org", false);
+            var request = new TestShowHotspotRequest("http://localhost/", "project", "any", "org");
+            var serverMock = new Mock<ISonarQubeService>();
+
+            IOpenInIDERequestHandler testSubject = new OpenInIDERequestHandler(validatorMock.Object, serverMock.Object, Mock.Of<ILogger>());
+
+            Action act = () => testSubject.ShowHotspot(request);
+
+            act.Should().NotThrow();
+            validatorMock.VerifyAll();
+            serverMock.Invocations.Should().BeEmpty();
+        }
+
+        private static Mock<IOpenInIDEStateValidator> CreateValidator(string uri, string projectKey, string orgKey, bool canHandleRequest)
+        {
+            var mockValidator = new Mock<IOpenInIDEStateValidator>();
+            mockValidator.Setup(x => x.CanHandleOpenInIDERequest(It.Is<Uri>(x => x.ToString() == uri), projectKey, orgKey))
+                .Returns(canHandleRequest);
+
+            return mockValidator;
+        }
+
+        private class TestShowHotspotRequest : IShowHotspotRequest
```

Comment:
Why not use a mock and setup the return values?

Reply:
No strong reason, except I find the mock Setup syntax clumsier and harder to read, and it isn't obvious whether the whole object is being mocked or not.